kartothek/serialization/_parquet.py

```diff
@@ -347,14 +348,20 @@ def _normalize_value(value, pa_type):
         elif isinstance(value, bytes):
             value = value.decode("utf-8")
             return datetime.datetime.strptime(value, "%Y-%m-%d").date()
-        elif isinstance(value, datetime.date) and not isinstance(
-            value, datetime.datetime
-        ):
-            return value
+        elif isinstance(value, datetime.date):
+            if isinstance(value, datetime.datetime):
+                raise TypeError(
```

Comment:
Special case for matching datetime predicate values against date columns. Generic error message below does not handle this nicely